src/Application/DiscordApplication.php

```diff
@@ -95,7 +114,24 @@ public function getUsers(): array
             throw new \RuntimeException('No guild was selected');
         }
 
-        return $this->userExtractor->extractForGuild($guildId);
+        $users = $this->userExtractor->extractForGuild($guildId);
+
+        if ($this->groups) {
+            // Store relation User <-> Group in Group model
+            foreach ($users as $user) {
+                $userGroupIds = $user->getExtra()['groups'] ?? [];
+
+                if ($userGroupIds) {
+                    foreach ($userGroupIds as $userGroupId) {
```

Comment:
I would use: `if ($userGroupIds) { continue; }` to avoid such nesting

BTW, is it possible to have `$user->getExtra()['groups'] === false` ?
if it's not possible, the  `if` becomes useless